db/upgrade.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package    tool_etl
+ * @copyright  2020 Tendai Mpita <tendaimpita@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die();
+
+function xmldb_tool_etl_upgrade($oldversion) {
+    global $DB;
+
+    if ($oldversion < 2020012401) {
+        $dtable = 'tool_etl_task';
+
+        // Get all tasks with all their settings
+        $tasks = $DB->get_records('tool_etl_task',  $conditions=null, $sort='', $fields='id, target, target_settings', $limitfrom=0, $limitnum=0);
```

Comment:
More for aesthetics, but generally I like to remove clutter that doesn't matter to improve readability:

$tasks = $DB->get_records('tool_etl_task',  $conditions=null, $sort='', $fields='id, target, target_settings', $limitfrom=0, $limitnum=0);

vs

$tasks = $DB->get_records('tool_etl_task',  null, null, 'id, target_settings');

You don't seem to use 'target'